src/reconcile/rec_visibility.c

```diff
@@ -134,30 +234,243 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
+            continue;
+
+        /*
+         * Lookaside without stable timestamp was taken care of above (set to the first uncommitted
+         * transaction). All other reconciliation takes the first stable update.
+         */
+        upd_select->upd = upd;
+        break;
+    }
+    WT_ERR(ret);
+
+    /*
+     * This means that we're choosing an update from lookaside for reconciliation AND that it is a
+     * modify. If that's the case then we need to expand the modify into a full standard update
+     * since modifies don't go on-disk.
+     *
+     * We might need to think about this a bit more... since the allocation of the modify update was
+     * a waste.
+     */
+    if (upd_select->upd != NULL && upd_select->upd->ext != 0 &&
+      upd_select->upd->type == WT_UPDATE_MODIFY) {
+        WT_CLEAR(las_key);
+        WT_CLEAR(las_value);
+        las_btree_id = 0;
+        las_ts = WT_TS_NONE;
+        las_txnid = WT_TXN_NONE;
+
+        /* Find the birthmark update if one exists in the update list. */
+        for (birthmark_upd = first_inmem_upd;
+             birthmark_upd != NULL && birthmark_upd->type != WT_UPDATE_BIRTHMARK;
+             birthmark_upd = birthmark_upd->next)
+            ;
+
+        /*
+         * We'll need to walk backwards accumulating modifies and then apply them to the next full
+         * update we come across.
+         */
+        WT_ERR(__wt_modify_vector_push(&modifies, upd_select->upd));
+        WT_ERR(
+          las_cursor->get_value(las_cursor, &durable_ts, &prepare_state, &upd_type, &las_value));
+        while (upd_type == WT_UPDATE_MODIFY) {
+            WT_ERR(__wt_update_alloc(session, &las_value, &tmp_upd, &notused, upd_type));
+            WT_ERR(__wt_modify_vector_push(&modifies, tmp_upd));
+            mod_upd_ts = tmp_upd->start_ts;
+            tmp_upd = NULL;
+
+            /*
+             * Getting "not found" means we've hit the beginning of the lookaside table and is the
+             * same thing as hitting a key boundary. It means that the base update that we're
+             * applying modifies on top of must be the on-disk value.
+             */
+            ret = las_cursor->prev(las_cursor);
+            WT_ERR_NOTFOUND_OK(ret);
+            cmp = 0;
+            if (ret != WT_NOTFOUND) {
+                WT_ERR(
+                  las_cursor->get_key(las_cursor, &las_btree_id, &las_key, &las_ts, &las_txnid));
+                /*
+                 * If the birthmark update is more recent than the current lookaside record then we
+                 * should use that as the base update instead.
+                 */
+                if (birthmark_upd != NULL && las_ts < mod_upd_ts &&
+                  ((birthmark_upd->start_ts > las_ts) ||
+                      (birthmark_upd->start_ts == las_ts && birthmark_upd->txnid > las_txnid))) {
+                    upd_type = WT_UPDATE_STANDARD;
+                    prepare_state = WT_PREPARE_INIT;
```

Comment:
@tetsuo-cpp , It looks like you are clearing out `upd_type` and `prepare_state` before breaking out of the loop, but it is not obvious to me why. Can you please explain and maybe put a comment as well.

Reply:
7c6c522

Setting `prepare_state` is unnecessary so I've removed that. The reason is because we use the update type as part of update allocation. Normally it gets set when we hit the base update in the lookaside traversal. This case is different since we're using the birthmark update so I set it explicitly. Also left a comment to that effect.